.github/workflows/unittest.yaml

```diff
@@ -0,0 +1,36 @@
+name: unittest
+
+on:
+  pull_request:
+    branches: [ main ]
+
+  workflow_dispatch:
```

Comment:
Can you add the appropriate `paths-ignore` here so that documentation only changes don't cause this to run?  You should be able to look at other workflows in this repo for examples.  Also, it will need a corresponding faux workflow.